source/includes/api/_query.md

```diff
@@ -1,27 +1,269 @@
+                }
+            }
+        }
+    }
 }
-'
 ```
 
+To execute a query over a dataset's data, you can either perform a GET request providing the SQL query as query param, or a POST request providing the SQL query in the request body.
+
+### Errors for querying datasets
+
+Error code     | Error message  | Description
+-------------- | -------------- | --------------
+400            | SQL o FS required | The required `sql` field is missing either as query string parameter or in the request body.
+
+## Download
+
+> Requesting the download of a query providing the dataset ID in the URL:
+
+```shell
+curl -X GET https://api.resourcewatch.org/v1/download/<dataset.id>?sql=SELECT * from <dataset.tableName>
+```
+
+> Requesting the download of a query providing the dataset ID in the FROM clause of the query:
+
 ```shell
-curl -i -XGET http\://api.resourcewatch.org/v1/query\?sql\=SELECT\ \*\ from\ <dataset.slug>
+curl -X GET https://api.resourcewatch.org/v1/download?sql=SELECT * from <dataset.id>
 ```
 
-<aside class="notice">
-While we aim to make our query interface as broad and transparent as possible, some of the querying options described 
-below will not be available for specific dataset providers, depending on this API's implementation or limitations on the
-actual data provider's side.
-</aside>
+> Explicitly setting the format of the returned file:
+
+```shell
+curl -X GET https://api.resourcewatch.org/v1/download?sql=SELECT * from <dataset.id>&format=json
+curl -X GET https://api.resourcewatch.org/v1/download?sql=SELECT * from <dataset.id>&format=csv
+```
 
-## Select clause
+You can download the result of a query using the `download` endpoint. You can either provide the dataset id in the URL or you can pass it as part of the SQL query (in the FROM clause).
+
+You can also specify which file type you want to download (JSON or CSV) - except for Google Earth Engine datasets (which only supports JSON). By default, the API will return a CSV file (or a JSON file for Google Earth Engine).
+
+**Please not that not all dataset providers support downloading queries - the following providers support downloading query results:**
+
+* Google Earth Engine
+* Document
+* Carto
+* BigQuery
+* ArcGIS FeatureService
+
+## Limitations of SQL syntax
+
+**While the WRI API aims to make the query interface as broad and transparent as possible, some of the querying options described below will not be available for specific dataset providers, depending on this API's implementation or limitations on the actual data provider's side.**
+
+Additionally to provider-specific limitations, every SQL query is transformed by [the `sql2json` microservice](https://github.com/resource-watch/sql2json), also maintained as [NPM package](https://www.npmjs.com/package/sql2json). There is a first conversion from SQL to JSON, and then from JSON to a SQL syntax that is compatible with [the Elasticsearch SQL plugin](https://github.com/NLPchina/elasticsearch-sql).
```

Comment:
I think this whole section should be in the dev docs, not these docs. End users don't really need to know about how we handle queries, we could just have a "internally, the API transforms your SQL queries to match the query syntax of the underlying dataset type"

Reply:
Also: this section seems to focus more on ES datasets, so maybe part of this info should be moved to the "Document-based datasets" section below?

Reply:
I think some of this info might be important to troubleshoot why a given query might not be returning the expected results (that is, the part under "This SQL syntax supported by Elasticsearch has some limitations that should be considered when querying datasets:") - so, I'll move that part to the "Document-based datasets" section below, and move the remaining (docs on how every SQL query is transformed) to the developer docs.